integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java

```diff
@@ -1,77 +1,261 @@
+                .then(this::verifyFreeIpaVolumeEncryptionKey)
+                .given(SdxTestDto.class)
+                    .withCloudStorage()
+                    .withExternalDatabase(sdxDatabaseRequest)
+                .when(sdxTestClient.create())
+                .await(SdxClusterStatusResponse.RUNNING)
+                .awaitForHealthyInstances()
+                .then(this::verifySdxVolumeEncryptionKey)
                 .given(DistroXTestDto.class)
-                .withInstanceGroupsEntity(distroXInstanceGroupTestDtos)
-                .withExternalDatabase(externalDatabaseName)
+                    .withInstanceGroupsEntity(distroXInstanceGroupTestDtos)
+                    .withExternalDatabase(distroXDatabaseRequest)
                 .when(distroXTestClient.create())
                 .await(STACK_AVAILABLE)
-                .then((tc, testDto, client) -> {
-                    sshEnaDriverCheckActions.checkEnaDriverOnAws(testDto.getResponse(), client);
-                    return testDto;
-                })
-                .then(new AvailabilityZoneAssertion())
+                .awaitForHealthyInstances()
+                .then(this::verifyAwsEnaDriver)
+                .then(new AwsAvailabilityZoneAssertion())
                 .then(validateTemplateContainsExternalDatabaseHostname())
-                .then((context, distrox, client) -> {
-                    distrox.getResponse();
-                    return distrox;
-                })
                 .validate();
     }
 
+    private DistroXTestDto verifyAwsEnaDriver(TestContext testContext, DistroXTestDto testDto, CloudbreakClient cloudbreakClient) {
+        if (CloudPlatform.AWS.equals(testDto.getCloudPlatform())) {
+            sshEnaDriverCheckActions.checkEnaDriverOnAws(testDto.getResponse(), cloudbreakClient);
+        } else {
+            LOGGER.warn(format("ENA driver is only available at AWS. So validation on '%s' provider is not possible!", testDto.getCloudPlatform()));
+        }
+        return testDto;
+    }
+
+    private EnvironmentTestDto verifyEnvironmentResponseDiskEncryptionKey(TestContext testContext, EnvironmentTestDto testDto,
+            EnvironmentClient environmentClient) {
+        DetailedEnvironmentResponse environment = environmentClient.getDefaultClient().environmentV1Endpoint().getByName(testDto.getName());
+        if (CloudPlatform.AWS.name().equals(environment.getCloudPlatform())) {
+            String encryptionKeyArn = environment.getAws().getAwsDiskEncryptionParameters().getEncryptionKeyArn();
+            environmentName = testDto.getRequest().getName();
+            verifyDiskEncryptionKey(testDto.getCloudPlatform(), encryptionKeyArn);
+        } else if (CloudPlatform.AZURE.name().equals(environment.getCloudPlatform())) {
+            String diskEncryptionSetId = environment.getAzure().getResourceEncryptionParameters().getDiskEncryptionSetId();
+            environmentName = testDto.getRequest().getName();
```

Comment:
I find the way the environment name gets extracted very error-prone. In particular, `environmentName` is an instance field, which means that it would be impossible to have multiple test cases in this class being run in parallel (were there multiple ones). For now it will work fine, as `testCreateDistroXWithEncryptedVolume()` is the sole test case, but the approach here is certainly not bulletproof.